### ISSUE TITLE
Minimap Feature List

### ISSUE THREAD

**Issue by felixjones** [caffeinefree/plugins#1]
Need to come up with features for the minimap plugin.

The minimap would be split into the Sprite class, for rendering the minimap, and the Window class, for displaying the minimap.

**Comment by TheUnproPro** [comment]
using Q/W (L/R on controller, pgdown/pgup according to the input method) to control how far you're zoomed in/out with the minimap.

Another nice feature would be press M to see the whole map. It'd also be nice if you only reveal parts of the map if you've walked near that area

**Comment by felixjones** [comment]
### Window Gravity

This is how Window_Minimap is positioned on the containing Scene. This should be resolution independent, so works for any screen size.

There's 9 positions:
1. upper-left
2. upper-middle
3. upper-right
4. middle-left
5. middle-middle
6. middle-right
7. bottom-left
8. bottom-middle
9. bottom-right

Along with this, an XY offset would be useful for adding a margin with the edge of the screen, or offsetting in the middle of the screen.

**Comment by TheUnproPro** [comment]
Agreed for the X/Y Offset. Perhaps an options menu similar to how I did before, only better and properly coded this time... lmao

**Comment by felixjones** [comment]
> Another nice feature would be press M to see the whole map.

I think this could be done by stretching the Window_Minimap to cover the entire screen, which means it could be animated to grow - animation speed would need to be configurable.

A new scene would be required. Perhaps Scene_Minimap. It could then allow the player to use the arrow keys to scroll the map and look around.

**Comment by felixjones** [comment]
Revealing visited locations of the map "fog of war" style would mean storing the revealed areas in the save data.

**Comment by TheUnproPro** [comment]
Yep, could also be pretty resource heavy, haven't a clue how to make it possible with ease unless we make it blocky

**Comment by felixjones** [comment]
Should we simplify how the minimap is styled?
I think two colors would be best; one for passable tiles and one for impassable tiles. Have these two colours configurable.

Another setting would be to configure the tiles to be colored based on the tileset, Minecraft style.

I also think a circular minimap should be an option.

**Comment by TheUnproPro** [comment]
We could avoid drawing both tiles and simply draw a background with impassable colors, this would improve performance (unless its needed for the way you have in mind)

**Comment by felixjones** [comment]
I think ideally we'd cache the minimap, so we'd draw it once and pan it about, which means performance for drawing isn't an issue.

That being said, clearing the background to the impassable color would be the best option.
